### ISSUE TITLE
Improve convergence diagnostic message for NA

### ISSUE THREAD

**Issue by avehtari** [stan-dev/rstan#769]
#### Summary:
Related to the discussion of warnings of Rhat and ESS being NA at Stan discourse https://discourse.mc-stan.org/t/largest-rhat-is-na-when-one-parameter-is-fixed-to-a-constant/15488/5
and the older discussion of behavior for constant draws
https://discourse.mc-stan.org/t/ess-rhat-behavior-for-chains-numerical-consistent-with-constants/10465

tl;dr for the above: When all draws for a parameter have the same value, we can't know whether it is due to 1) the parameter being fixed, 2) limited floating point accuracy, 3) for a discrete quantity one state has high probability and the chain is not visiting the other states, 4) chains are stuck.  Due to the ambiguity rhat and ess functions return in these cases NA. 

It would be helpful to separately report if rhat and ess are bad for NA cases and separately report if there NAs.

Now the warnings are 
```
Warning messages:
1: The largest R-hat is NA, indicating chains have not mixed.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#r-hat 
2: Bulk Effective Samples Size (ESS) is too low, indicating posterior means and medians may be unreliable.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#bulk-ess 
3: Tail Effective Samples Size (ESS) is too low, indicating posterior variances and tail quantiles may be unreliable.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#tail-ess 
```
They could be like
```
Warning messages:
1: Some parameters have constant draws which can be due to parameters being constant, limited numerical accuracy or mixing problems. R-hat, Bulk-ESS and Tail-ESS for these parameters will be reported as NA.
See http://mc-stan.org/misc/warnings.html#r-hat 
2: The largest (non NA) R-hat is x.x, indicating chains have not mixed.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#r-hat 
3: The smallest (non-NA) Bulk Effective Samples Size (ESS) is y.y, indicating posterior means and medians may be unreliable.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#bulk-ess 
4: The smallest (non-NA) Tail Effective Samples Size (ESS) is z.z, indicating posterior variances and tail quantiles may be unreliable.
Running the chains for more iterations may help. See
http://mc-stan.org/misc/warnings.html#tail-ess 
```

**Comment by jgabry** [comment]
I like these proposed changes. @bgoodri and @paul-buerkner (and anyone else) what do you think? Anyone have any problem with changing these warning messages?

**Comment by bgoodri** [comment]
I think I just added a na.rm = TRUE on GitHub, but if we want to change the
warning messages to add detail like this do it right now because we might
have a brief window to get rstan 2.21.x onto CRAN before something else
breaks.

**Comment by jgabry** [comment]
Just seeing this. na.rm=T is fine for now. We can change the messages next time when we have more time if you want to go ahead and submit rstan. let me know when I need to submit rstantools. I’m leaving shortly for my trip but I’ll find time to submit rstantools whenever you give me the go-ahead

**Comment by bgoodri** [comment]
Have you run R CMD check on the reverse dependencies of rstantools 2.1.1
recently?

**Comment by jgabry** [comment]
Not in the past two weeks or so. I’m on my phone right now and won’t be able to run anything until tonight at the earliest. If you want to run rstantools reverse dependency checks go for it, otherwise I can try running them tonight but not sure what time that will be, could be late.

**Comment by bgoodri** [comment]
tonight is fine

**Comment by paul-buerkner** [comment]
I think removing NA for now is fine and we can make better warnings when moving the warnings to posterior, as is currently planed I think.

**Comment by avehtari** [comment]
I assume NA was removed? I'll work with Paul on warning messages in the beginning of August.

**Comment by jgabry** [comment]
Yeah looks like Ben removed them in https://github.com/stan-dev/rstan/commit/54de3d26c080cd9d266104a33eeedb089c0234d2
